### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by cthulahoops**
Add basic support for classes.

Working:
* Define classes.
* Instantiate classes.
* Set attributes on classes and instances.

Not working:
* Access class attributes via instances.
* Invoke methods.
* Inheritance.
* Meta-classes.

Type Kind is probably wrong, and will probably go away later.

### CODE REVIEW

File: vm/src/compile.rs [hunk 1]
```diff
@@ -318,8 +351,14 @@ impl Compiler {
                 self.compile_expression(b);
                 self.emit(Instruction::StoreSubscript);
             }
+            ast::Expression::Attribute { value, name } => {
+                self.compile_expression(value);
+                self.emit(Instruction::StoreAttr {
```
Comment (windelbouwman) [inline vm/src/compile.rs]: This should be Instruction::get_attr, not storeAttr right? When we face an attribute expression, we must fetch the attribute.

File: vm/src/vm.rs [hunk 2]
```diff
@@ -460,11 +478,16 @@ impl VirtualMachine {
 
     fn new_instance(&mut self, type_ref: PyObjectRef, args: PyFuncArgs) -> PyResult {
         // more or less __new__ operator
-        let obj = PyObject::new(PyObjectKind::None, type_ref.clone());
+        let dict = self.new_dict();
+        let obj = PyObject::new(PyObjectKind::Instance{dict: dict}, type_ref.clone());
+        let init = type_ref.get_attr(&String::from("__init__"));
+        let mut self_args = PyFuncArgs { args: args.args };
+        self_args.args.insert(0, obj.clone());
+        self.invoke(init, self_args).unwrap();
```
Comment (windelbouwman) [inline vm/src/vm.rs]: Here, the result must be checked, and if an exception is there, it must be returned.

File: vm/src/compile.rs (hunk 1 above)
Comment (cthulahoops) [reply vm/src/compile.rs]: That's inside compile_store.

File: vm/src/vm.rs (hunk 2 above)
Comment (cthulahoops) [reply vm/src/vm.rs]: I'll push a fix to master.

**Pull request merged**